contracts/vendor/proxy/EIP173ProxyWithReceive.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.11;
+
+import "./EIP173Proxy.sol";
+
+///@notice Proxy implementing EIP173 for ownership management that accept ETH via receive
+contract EIP173ProxyWithReceive is EIP173Proxy {
+    constructor(
+        address implementationAddress,
+        address ownerAddress,
+        bytes memory data
+    ) payable EIP173Proxy(implementationAddress, ownerAddress, data) {}
+
+    receive() external payable override {
```

Comment:
maybe name this contract `EIP173ProxyWithCustomReceive` and include custom receive functionalilty (delegate call to implementationAddress)

Reply:
If so, remember to change `proxyContract` in `deploy/TaskTreasuryUpgradable.deploy.js` to new contract name

Reply:
This custom  `EIP173ProxyWithReceive` is being used in `deploy/TaskTreasuryUpgradable.deploy.js` instead of the ones from hardhat deploy. Do u mean i should rename it?

Reply:
renamed to `EIP173ProxyWithCustomReceive` in c1056a1 to make it clearer